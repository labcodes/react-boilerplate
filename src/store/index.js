import { applyMiddleware, createStore } from 'redux';
import logger from 'redux-logger';

import actions from '../actions';

// ====

const store = createStore(
    actions,
    applyMiddleware(logger)
);

// ====

export default store;
