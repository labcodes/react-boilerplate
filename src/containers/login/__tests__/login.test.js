'use strict';

import React from 'react';

// Test helpers
import renderer from 'react-test-renderer';

// Containers connected
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from 'store';

// Component
import Login from 'containers/login';

// ====

describe('Login [Snapshot]', () => {
    it('Renders correctly', () => {
        const component = (
            <BrowserRouter>
                <Provider store={store}>
                    <Login />
                </Provider>
            </BrowserRouter>
        );

        const tree = renderer.create(component).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
