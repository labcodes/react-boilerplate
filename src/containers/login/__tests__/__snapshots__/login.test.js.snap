// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login [Snapshot] Renders correctly 1`] = `
Array [
  <section>
    <h1>
        Login!
    </h1>
    <form
        onSubmit={[Function]}
    >
        <div
            className="field"
        >
            <label
                className="label"
                htmlFor="Username"
            >
                Username
            </label>
            <div
                className="control"
            >
                <input
                    autoComplete="off"
                    className="input"
                    name="username"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onFocus={[Function]}
                    placeholder="your username"
                    type="text"
                    value=""
                />
            </div>
        </div>
        <div
            className="field"
        >
            <label
                className="label"
                htmlFor="E-mail"
            >
                E-mail
            </label>
            <div
                className="control"
            >
                <input
                    autoComplete="off"
                    className="input"
                    name="e-mail"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onFocus={[Function]}
                    placeholder="your email"
                    type="email"
                    value=""
                />
            </div>
        </div>
        <div
            className="field"
        >
            <label
                className="label"
                htmlFor="UF"
            >
                UF
            </label>
            <div
                className="control"
            >
                <input
                    autoComplete="off"
                    className="input"
                    name="uf"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onDragStart={[Function]}
                    onDrop={[Function]}
                    onFocus={[Function]}
                    placeholder="your uf code"
                    type="text"
                    value=""
                />
            </div>
        </div>
        <div
            className="field is-grouped"
        >
            <div
                className="control"
            >
                <button
                    className="button is-text"
                    disabled={true}
                    onClick={[Function]}
                    type="button"
                >
                    Cancel
                </button>
            </div>
            <div
                className="control"
            >
                <button
                    className="button is-link"
                    disabled={true}
                    type="submit"
                >
                    Submit
                </button>
            </div>
        </div>
    </form>
</section>,
  <section>
    <h1>
        Fake data!
    </h1>
    <button
        onClick={[Function]}
    >
        load
    </button>
</section>,
  <section>
    <h1>
        Remote submit!
    </h1>
    <div>
        <button
            onClick={[Function]}
        >
            submit
        </button>
        <button
            onClick={[Function]}
        >
            reset
        </button>
    </div>
</section>,
]
`;
